simde/x86/avx512f.h

```diff
@@ -1243,6 +1295,59 @@ simde_mm512_setone_pd(void) {
   return simde_mm512_castsi512_pd(simde_mm512_setone_si512());
 }
 
+SIMDE__FUNCTION_ATTRIBUTES
+simde__m512i
+simde_mm512_srli_epi64 (simde__m512i a, unsigned int imm8) {
+#if defined(SIMDE_AVX512F_NATIVE)
+  return _mm512_srli_epi64(a, imm8);
+#else
+  simde__m512i_private
+    r_,
+    a_ = simde__m512i_to_private(a);
+
+#if defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
+  r_.u64 = a_.u64 >> HEDLEY_STATIC_CAST(int32_t, imm8);
```

Comment:
The cast here is unnecessary, but the [documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_srli_epi64&expand=5530) says this function only looks at the least significant 8 bits, so you'll want to mask `0xff`.

You might want to use `int_fast16_t` for the result of the mask operation (in which case you would need a cast).  I would say `int_fast8_t`, but that's actually `sizeof(char)` in a lot of places (including my x86_64 desktop).  That should give the compiler the most room to choose the fastest type.

Reply:
Also, to be clear, the loop below suffers from the same issue, so the mask should be common to both cases.

Reply:
OK, I'll correct.
To me all this HEDLEY stuff is a mystery.
What is its purpose?
The function I pinched the code from had this HEDLEY_STATIC_CAST so I thought it was necessary.

Reply:
[Hedley](https://github.com/nemequ/hedley) doesn't really have one purpose; it's a collection of stuff to do different things.  You might want to skim through [the web site](https://nemequ.github.io/hedley/).

`HEDLEY_STATIC_CAST` is pretty straightforward; in C++, the code you have above gets preprocessed to `static_cast<uint32_t>(imm8)`, and in C it gets preprocessed to `((uint32_t) imm8)`.  Basically, it helps keep C++ compilers happy while preserving compatibility with C.  Several C++ compilers actually have a warning (`-Wold-style-cast` in clang, I know IAR has one, too) if you use a C-style cast in your C++ code.